src/IdempotentAPI.AccessCache/Lockers/InProcessAccessLock.cs

```diff
@@ -15,44 +17,53 @@ namespace IdempotentAPI.AccessCache.Lockers
     /// </example>
     internal sealed class InProcessAccessLock : IDisposable
     {
-        private static readonly object localLocker = new object();
-        private static readonly Dictionary<string, LockDisposeTracker> lockValueMapper = new Dictionary<string, LockDisposeTracker>();
+        private static readonly ConcurrentDictionary<string, LockDisposeTracker> lockValueMapper = new ();
 
         private readonly string _lockedValue;
 
         public InProcessAccessLock(string valueToLock)
         {
             _lockedValue = valueToLock;
+        }
 
-            LockDisposeTracker disposeTracker;
-            lock (localLocker)
+        public static async Task<InProcessAccessLock> CreateAsync(string valueToLock)
+        {
+            var disposeTracker = lockValueMapper.AddOrUpdate(valueToLock, (_) => new LockDisposeTracker(), (_, disposeTracker) => disposeTracker with
             {
-                if (!lockValueMapper.TryGetValue(valueToLock, out disposeTracker))
-                {
-                    disposeTracker = new LockDisposeTracker();
-                    lockValueMapper.Add(valueToLock, disposeTracker);
-                }
-                disposeTracker.Count++;
-            }
-            Monitor.Enter(disposeTracker);
+                Count = disposeTracker.Count + 1
+            });
+            await disposeTracker.Semaphore.Value.WaitAsync();
+            return new InProcessAccessLock(valueToLock);
```

Comment:
It was needed to rewrite in-proc lock to use `SemaphoreSlim` because `Monitor` does not support async flow

Reply:
Thank you, @dimmy-timmy for your effort ❤️

Reply:
Modifying the integration tests to use WebApplicationFactory is an excellent addition 💪 . I am removing it from my TODO list 😄